File: apps/marrow/backend/app/schemas/insights.py
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, ConfigDict


class TrendPoint(BaseModel):
    date: str
    value: Optional[float]
    rolling_avg_7: Optional[float]


class TrendSeries(BaseModel):
    key: str
    label: str
    category: str
    points: list[TrendPoint]
    current: Optional[float]
    rolling_avg_7: Optional[float]
    delta_30d: Optional[float]


class TrendsResponse(BaseModel):
    series: list[TrendSeries]


class TreatmentResponseRow(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    treatment_id: int
    name: str
    type: str
    start_date: str
    end_date: Optional[str]
    baseline_mean: Optional[float]
    during_mean: Optional[float]
    after_mean: Optional[float]
    baseline_n: int
    during_n: int
    after_n: int
    delta_during_vs_baseline: Optional[float]


class TreatmentResponseList(BaseModel):
    outcome: str
    rows: list[TreatmentResponseRow]

File: apps/marrow/frontend/lib/api/types/insights.ts
export interface TrendPoint {
  date: string
  value: number | null
  rolling_avg_7: number | null
}

export interface TrendSeries {
  key: string
  label: string
  category: string
  points: TrendPoint[]
  current: number | null
  rolling_avg_7: number | null
  delta_30d: number | null
}

export interface TrendsResponse {
  series: TrendSeries[]
}

export interface TreatmentResponseRow {
  treatment_id: number
  name: string
  type: string
  start_date: string
  end_date: string | null
  baseline_mean: number | null
  during_mean: number | null
  after_mean: number | null
  baseline_n: number
  during_n: number
  after_n: number
  delta_during_vs_baseline: number | null
}

export interface TreatmentResponseList {
  outcome: string
  rows: TreatmentResponseRow[]
}
